compose-builder/Makefile

```diff
@@ -130,14 +156,31 @@ ifeq (ds-random, $(filter ds-random,$(ARGS)))
 		else
 			TOKEN_LIST:=$(TOKEN_LIST),device-random
 		endif
+		ifeq ($(KNOWN_SECRETS_LIST),"")
+			KNOWN_SECRETS_LIST:=redisdb[device-random]
+		else
+			KNOWN_SECRETS_LIST:=$(KNOWN_SECRETS_LIST),redisdb[device-random]
+		endif
 		extension_file:= $(shell GEN_EXT_DIR="$(GEN_EXT_DIR)" ./gen_secure_compose_ext.sh device-random)
 		COMPOSE_FILES:=$(COMPOSE_FILES) -f $(extension_file)
 	endif
 endif
 ifeq (ds-rest, $(filter ds-rest,$(ARGS)))
 	COMPOSE_FILES:=$(COMPOSE_FILES) -f add-device-rest.yml
 	ifneq (no-secty, $(filter no-secty,$(ARGS)))
-		extension_file:= $(shell GEN_EXT_DIR="$(GEN_EXT_DIR)" ./gen_secure_compose_ext.sh device-rest device-rest device-rest-go)
+		# TODO: Remove setting token list once device rest's service key has been changed
+		#       to 'device-rest' or edgex-go has been update to use 'edgex-device-rest'.
```

Comment:
just a note: right now, edgex-go has both `device-rest`  and `device-virtual` as it is and changing device-virtual to edgex-device-virtual means to change all token paths as well, including those in snap package building.